config/locales/en.yml

```diff
@@ -23,3 +23,7 @@ en:
   hello: "Hello world"
   sets:
     set_not_found: "Invalid set requested."
+  hydra_editor:
+    form:
+      model_label:
+        GenericSet: "Collection"
```

Comment:
I don't think we want to change the label. `Collection` has meaning in libraries that we don't want to confuse.